sample_apps/run_example.sh

```diff
@@ -241,7 +352,28 @@ function emded_policy_in_example_app() {
 }
 
 # ###########################################################################
+# As this function is executed through a table of function names,
+# caller cannot pass-in the sample-app-name. So, fall back to using
+# a global variable.
 function compile_app() {
+    case "${SampleAppName}" in
```

Comment:
You will see this pattern in many such functions.

The entry-point. step function, `compile_app()`, farms off control to application-specific step function based on the sample-app-name established at the beginning of the script's execution.